helix-core/src/test/java/org/apache/helix/controller/rebalancer/waged/TestAssignmentMetadataStore.java

```diff
@@ -81,7 +90,7 @@ public void beforeClass() throws Exception {
     _manager.connect();
 
     // create AssignmentMetadataStore
-    _store = new AssignmentMetadataStore(_manager.getMetadataStoreConnectionString(),
+    _store = new MockAssignmentMetadataStore(_manager.getMetadataStoreConnectionString(),
```

Comment:
Why we need the mock store here? If it is not testing against the original class, this test case won't be valid, right? Imaging someone modifies the real store class without touching this mock, the test case will still pass.

I guess you create this mock class for 2 reasons:
1. To expose the internal field. For this, we can create some protected methods that can be used to access these private fields.
2. To add the counters. For this, we might have to introduce a child class, but instead of re-write the whole method logic, we should at least call super.persistBaseline(), for example, to ensure the origin class'es method logic is tested here.

Reply:
Moreover, I think even we need this mock, it should be used for testAvoidingRedundantWrite() only. As for the other test method, they should be still using the original assignment store class.